ui/components/multichain/network-list-menu/network-list-menu.tsx

```diff
@@ -291,78 +316,147 @@ export const NetworkListMenu = ({ onClose }: { onClose: () => void }) => {
       setNetworkClientIdForDomain(selectedTabOrigin, networkClientId);
     }
 
+    if (permittedAccountAddresses.length > 0) {
+      dispatch(addPermittedChain(selectedTabOrigin, hexChainId));
+      if (!permittedChainIds.includes(hexChainId)) {
+        dispatch(showPermittedNetworkToast());
+      }
+    }
+  };
+
+  const handleNonEvmNetworkChange = async (chainId: CaipChainId) => {
+    if (isAccountInNetwork(chainId)) {
+      dispatch(toggleNetworkMenu());
+      dispatch(setActiveNetwork(chainId));
+      return;
+    }
+
+    setSelectedNonEvmNetwork(chainId);
+    setActionMode(ACTION_MODE.ADD_NON_EVM_ACCOUNT);
+  };
+
+  const handleNetworkChange = async (chainId: CaipChainId) => {
+    const { isEvm } = multichainNetworks[chainId];
+    if (isEvm) {
+      handleEvmNetworkChange(chainId);
+    } else {
+      await handleNonEvmNetworkChange(chainId);
+    }
+
+    const chainIdToTrack = isEvm ? convertCaipToHexChainId(chainId) : chainId;
+    const currentChainIdToTrack = isEvm
+      ? convertCaipToHexChainId(currentChainId)
+      : currentChainId;
```

Comment:
Very very nit: (I did not lint properly this 🙈)

```suggestion
    const [chainIdToTrack, currentChainIdToTrack] = isEvm
      ? [
        convertCaipToHexChainId(chainId), 
        convertCaipToHexChainId(currentChainId)
      ]
      : [chainId, currentChainId];
```

Reply:
Done in https://github.com/MetaMask/metamask-extension/pull/30433/commits/198ae2e8a3e00b9da396549ad9bbdc362181da70